### ISSUE TITLE
Local plugins can't be "ignore"'d with require hook

### ISSUE THREAD

**Issue by brigand** [babel/babel#2474]
When using a plugin that's not in node_modules it tries to transform itself. It can't be ignored in .babelrc

```
TypeError: The plugin "./babel-plugin-noop.js" didn't export a Plugin instance
    at PluginManager.validate (/usr/local/lib/node_modules/babel/node_modules/babel-core/lib/transformation/file/plugin-manager.js:164:13)
    at PluginManager.add (/usr/local/lib/node_modules/babel/node_modules/babel-core/lib/transformation/file/plugin-manager.js:213:10)
    at File.buildTransformers (/usr/local/lib/node_modules/babel/node_modules/babel-core/lib/transformation/file/index.js:237:21)
    at new File (/usr/local/lib/node_modules/babel/node_modules/babel-core/lib/transformation/file/index.js:139:10)
    at Pipeline.transform (/usr/local/lib/node_modules/babel/node_modules/babel-core/lib/transformation/pipeline.js:164:16)
    at Object.transformFileSync (/usr/local/lib/node_modules/babel/node_modules/babel-core/lib/api/node.js:137:37)
    at compile (/usr/local/lib/node_modules/babel/node_modules/babel-core/lib/api/register/node.js:132:20)
    at normalLoader (/usr/local/lib/node_modules/babel/node_modules/babel-core/lib/api/register/node.js:199:14)
    at Object.require.extensions.(anonymous function) [as .js] (/usr/local/lib/node_modules/babel/node_modules/babel-core/lib/api/register/node.js:216:7)
    at Module.load (module.js:356:32)
```

repro: https://gist.github.com/brigand/bacffb5e3dba750a1a2a

**Comment by jmm** [comment]
@brigand Just realized this is a dupe of @impinball's #2443. If you agree, can you please close this issue and we'll track it over there? Thanks! I do like your minimal repro though.

**Comment by brigand** [comment]
Closing as dupe.

**Comment by jmm** [comment]
Thanks @brigand! (Obviously I didn't realize it was a dupe either when it came up on slack yesterday.)

**Comment by loganfsmyth** [comment]
I'm actually going to reopen this in favor of #2443 since I think it's a clearer repro case and it has gotten a little convoluted in that other issue.

**Comment by impinball** [comment]
@loganfsmyth Thanks. This is worded a little clearer than mine, too. And it's because it tries to transform itself that's the bug here.

**Comment by idris** [comment]
:+1: I'm also running into #2443

**Comment by idris** [comment]
Has anyone found a cleaner workaround? Or is `babel-relay-plugin-loader` still the way to go?

**Comment by isiahmeadows** [comment]
@idris Placing it into a local `node_modules` folder (not at the root, obviously) is a decent workaround in my experience. I'd rather this fixed, but that workaround has been helpful for a while.

**Comment by hzoo** [comment]
This should be fixed in 7, can try `babel-register@7.0.0-alpha.4`
